rustup-init.ps1

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env powershell
+
+# This is just a little script that can be downloaded from the internet to
+# install rustup. It just does platform detection, downloads the installer and
+# runs it.
+
+#XXX: If you change anything here, please make the same changes in setup_mode.rs
+function Write-Usage {
+    Write-Host @'
+rustup-init 1.18.3 (435397f48 2019-05-22)
+The installer for rustup
+
+USAGE:
+    rustup-init.exe [FLAGS] [OPTIONS]
+
+FLAGS:
+    -v, --verbose           Enable verbose output
+    -y                      Disable confirmation prompt.
+        --no-modify-path    Don't configure the PATH environment variable
+    -h, --help              Prints help information
+    -V, --version           Prints version information
+
+OPTIONS:
+        --default-host <default-host>              Choose a default host triple
+        --default-toolchain <default-toolchain>    Choose a default toolchain to install
+'@
+}
+
+function main([AllowEmptyCollection()] [string[]]$Argv = @()) {
+    foreach ($arg in $Argv) {
+        if ("$arg" -eq "-h" -or "$arg" -eq "--help") {
+            Write-Usage
+            return
+        }
+    }
+
+    Install-Rustup $Argv
+}
+
+function Get-Architecture {
+    $platform = [System.Environment]::OSVersion.Platform
+    switch -Wildcard ($platform) {
+        # Linux and Darwin both report as Unix
+        "Unix" {
+            $ostype = uname -s
+            $cputype = uname -m
+            break
+        }
+        "Win*" {
+            $ostype = (Get-WmiObject Win32_OperatingSystem).Caption
```

Comment:
Get-WMI is not available in PowerShell Core.
https://docs.microsoft.com/en-us/powershell/scripting/whats-new/breaking-changes-ps6?view=powershell-6#wmi-v1-cmdlets

Use `Get-CimInstance` instead

Reply:
I should mention that I was hoping to support several older versions of PowerShell on the Windows side if it was trivial enough. The `Get-CimInstance` was one situation where I wasn't sure what version of PowerShell first shipped with `Get-CimInstance` vs. the older `Get-WmiObject`. Thank you for that link--is there a document or set of docs that would spell out these portability questions?

Reply:
Given my comments below about using the xplat variables, this WMI call is probably unnecessary i.e.

`($null -eq $IsWindows) -or $IsWindows)` is enough to detect if you're on a Windows platform.

Reply:
The whole `Get-Architecture` reads much better now and is a fraction of the size, thank you! I also learned that there is an implicit return in these functions which feels very Rust-y.

Reply:
> I also learned that there is an implicit return

Yeah, sorta-kinda. PowerShell is about the pipeline, so _ANY_ output generated by the function is put onto the pipeline, not just the last statement ala Ruby, Rust.

This can trip you up when when later, but fortunately not in this script.